src/sql/parts/notebook/models/clientSession.ts

```diff
@@ -70,20 +70,22 @@ export class ClientSession implements IClientSession {
 			await this.initializeSession();
 			await this.updateCachedKernelSpec();
 		} catch (err) {
-			this._errorMessage = notebookUtils.getErrorMessage(err);
+			this._errorMessage = notebookUtils.getErrorMessage(err) || localize('clientSession.unknownError', 'An error occurred while starting the notebook session');
 		}
 		// Always resolving for now. It's up to callers to check for error case
 		this._isReady = true;
 		this._ready.resolve();
-		this._kernelChangeCompleted.resolve();
+		if (!this.isInErrorState && this._session && this._session.kernel) {
+			await this.notifyKernelChanged(undefined, this._session.kernel);
+		}
 	}
 
 	private async startServer(): Promise<void> {
 		let serverManager = this.notebookManager.serverManager;
 		if (serverManager && !serverManager.isStarted) {
 			await serverManager.startServer();
 			if (!serverManager.isStarted) {
-				throw new Error(nls.localize('ServerNotStarted', 'Server did not start for unknown reason'));
+				throw new Error(localize('ServerNotStarted', 'Server did not start for unknown reason'));
```

Comment:
Minor: Key should be in camel case => serverNotStarted

Reply:
You told me to use double "" in localize msg:)

Reply:
Fixed in next iteration, thanks 😄